UiPath.FreeRdpClient/UiPath.FreeRdpClient/FreeRdpClient.cs

```diff
@@ -69,49 +30,47 @@ public static async Task<IAsyncDisposable> Connect(RdpConnectionSettings connect
             Port = connectionSettings.Port ?? default
         };
 
-        return await Task.Run(() =>
+        return await Task.Run(async () =>
         {
-            RdpLogon(connectOptions, out var releaseObjectName);
-            return new AsyncDisposable(() => { Disconnect(releaseObjectName); return ValueTask.CompletedTask; });
+            NativeInterface.RdpLogon(connectOptions, out var releaseObjectName);
+            return new AsyncDisposable(() =>
+            {
+                Disconnect(releaseObjectName); 
+                return ValueTask.CompletedTask;
```

Comment:
I have two concerns here:
- Dispose and DisposeAsync should never throw.
- If Task-like returning method throws, it should return a faulted task instead of failing to return one.

I would suggest we do at least the following:

```csharp
return new AsyncDisposable(() =>
{
    try 
    {
        Disconnect(releaseObjectName); 
        return ValueTask.CompletedTask;
    }
    catch (Exception ex)
    {
       return ValueTask.FromException(ex);
    }
});
```

### Explanation

In the callee, we do:

```cpp
...
if (!SetEvent(eventHandle))
{
    auto lastError = GetLastError();
    CloseHandle(eventHandle);
    return HRESULT_FROM_WIN32(lastError);
}
...
```

which through PreserveSig=false would lead to `Disconnect(releaseObjectName)` throwing.